cpp/include/cudf/lists/detail/scatter.cuh

```diff
@@ -526,10 +526,7 @@ struct list_child_constructor {
 
     if (num_child_rows == 0) {
       // make an empty lists column using the input child type
-      return make_empty_lists_column(
-        source_lists_column_view.child().child(lists_column_view::child_column_index).type(),
-        stream,
-        mr);
+      return empty_like(source_lists_column_view.child());
```

Comment:
Would be nice to add a scatter test case while were at it, since the need for this fix was due to a missed corner-case in testing.

Reply:
Sorry it took me a bit to find a test that triggered this and I finally had to go back to the original API that was causing the problem (partition).